include/dynd/kernels/string_split_kernel.hpp

```diff
@@ -0,0 +1,82 @@
+//
+// Copyright (C) 2011-15 DyND Developers
+// BSD 2-Clause License, see LICENSE.txt
+//
+
+// String split kernel
+
+#pragma once
+
+#include <dynd/string.hpp>
+#include <dynd/string_search.hpp>
+#include <dynd/types/var_dim_type.hpp>
+
+namespace dynd {
+  namespace nd {
+
+    struct string_split_kernel
+      : base_kernel<string_split_kernel, 2> {
+
+      typedef string_split_kernel self_type;
+
+      memory_block_data *m_dst_memblock;
+
```

Comment:
Rather then a `memory_block_data *`, let's store this as a `intrusive_ptr<memory_block_data>`, which is exactly what it is in the var_dim arrmeta. That gives us reference counting, but the semantics should otherwise be the same.